src/v/cloud_storage/remote_partition.h

```diff
@@ -91,6 +91,8 @@ class remote_partition
       storage::log_reader_config config,
       std::optional<model::timeout_clock::time_point> deadline = std::nullopt);
 
+    static size_t reader_mem_use_estimate() noexcept;
```

Comment:
nit: maybe name this `estimate_reader_mem_use` to match the other `estimate_` functions